cubed/array_api/dtypes.py

```diff
@@ -86,3 +87,35 @@
     "complex floating-point": _complex_floating_dtypes,
     "floating-point": _floating_dtypes,
 }
+
+
+# A Cubed-specific utility.
+def _validate_and_define_dtype(x, dtype=None, *, allowed_dtypes=("numeric",), fname=None, device=None):
+    """Ensure the input dtype is allowed. If it's None, provide a good default dtype."""
```

Comment:
Can we rename this something more concrete, like `_upcast_integral_dtypes`? It's used by `sum` and `prod` to ensure that the integral type is wide enough to avoid overflow (see dtype param doc for [`xp.sum`](https://data-apis.org/array-api/latest/API_specification/generated/array_api.sum.html#array_api.sum), and similarly for `xp.prod`).